R/xts.R

```diff
@@ -8,7 +8,7 @@ as.data.table.xts = function(x, keep.rownames = TRUE, key=NULL, ...) {
   if (identical(keep.rownames, FALSE)) return(r[])
   index_nm = if (is.character(keep.rownames)) keep.rownames else "index"
   if (index_nm %chin% names(x)) stop(domain=NA, gettextf("Input xts object should not have '%s' column because it would result in duplicate column names. Rename '%s' column in xts or use `keep.rownames` to change the index column name.", index_nm, index_nm))
-  r[, c(index_nm) := zoo::index(x)]
+  r[, c(index_nm) := zoo::index(x), env=list(x=x)]
```

Comment:
@mattdowle Great to use that you are using new feature. For future I would suggest to apply some styling preferences that will more easily convey which variables are substituted. So far I used to just prefix them with dot.
```r
r[, c(index_nm) := zoo::index(.x), env=list(.x=x)]
```
but I am open for any other alternative. It is not really needed because there is no collision, but for readability it could be useful.

Reply:
Yes this is a downside of this approach: the repetition of the variable names.  Using `.x` actually means more like `..x` already does and the . vs .. could be confusing; i.e. `.` conveys current level, and `..` conveys one level up, but not here where `env=` is used if we follow this single `.` prefix style.